scripts/s3/migrate_folder_language.py

```diff
@@ -0,0 +1,100 @@
+import logging
+import sys
+import time
+
+from modularodm import Q
+
+from framework.mongo import database
+from framework.transactions.context import TokuTransaction
+
+from website.app import init_app
+from website.addons.base.exceptions import InvalidAuthError, InvalidFolderError
+from website.addons.s3.utils import get_bucket_location_or_error
+from website.addons.s3.settings import BUCKET_LOCATIONS
+from website.addons.s3.model import S3NodeSettings
+from scripts import utils as script_utils
+
+
+logger = logging.getLogger(__name__)
+
+
+def migrate(dry_run=False):
+    bucket_name_location_map = {}
+    sns_collection = database['s3nodesettings']
+    logger.info('Migrating all {} S3NodeSettings documents: {}'.format(
+        sns_collection.count(), [s['_id'] for s in sns_collection.find()]
+    ))
+    for document in sns_collection.find():
+        sns_collection.find_and_modify(
+            {'_id': document['_id']},
```

Comment:
Currently, the entire migration is rolled back in the event of a failure (one caused by an exception, at least), so this is technically "respectful of multiple runs".

>would it be feasible or desirable to make it also check for documents where bucket is set or anything?

Yes and no, respectively. What this loop is really doing is closer to the rename of a database column... but the only documented indication that pymongo could rename a column proved unsuccessful upon manual testing. Something like 
```python
list(sns_collection.find({'$eq': {'bucket': None}})) + list(sns_collection.find({'$ne': {'bucket': None}}))
```
would probably return the actual set of things that need to be iterated over... but the chances of this mattering are close to zero.